Product/Production/Services/DocumentSubmissionCore/src/main/java/gov/hhs/fha/nhinc/docsubmission/audit/transform/DocSubmissionDeferredRequestAuditTransforms.java

```diff
@@ -0,0 +1,112 @@
+ *     * Neither the name of the United States Government nor the
+ *       names of its contributors may be used to endorse or promote products
+ *       derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE UNITED STATES GOVERNMENT BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package gov.hhs.fha.nhinc.docsubmission.audit.transform;
+
+import com.services.nhinc.schema.auditmessage.AuditMessageType;
+import gov.hhs.fha.nhinc.common.nhinccommon.AssertionType;
+import gov.hhs.fha.nhinc.docsubmission.audit.DocSubmissionAuditTransformsConstants;
+import gov.hhs.healthit.nhin.XDRAcknowledgementType;
+import ihe.iti.xds_b._2007.ProvideAndRegisterDocumentSetRequestType;
+
+/**
+ *
+ * @author tjafri
+ */
+public class DocSubmissionDeferredRequestAuditTransforms extends AbstractDocSubmissionAuditTransforms<
+    ProvideAndRegisterDocumentSetRequestType, XDRAcknowledgementType> {
+
+    @Override
+    protected AuditMessageType getParticipantObjectIdentificationForRequest(
+        ProvideAndRegisterDocumentSetRequestType request, AssertionType assertion, AuditMessageType auditMsg) {
+        // PatientParticipantObjetIdentification  and SubmissionSetParticipantObjetIdentification is same for both
+        //Request and Response in case of DS
+        auditMsg = getPatientParticipantObjectIdentification(request, auditMsg);
+        auditMsg = getSubmissionSetParticipantObjectIdentification(request, auditMsg);
+        return auditMsg;
+    }
+
+    @Override
+    protected AuditMessageType getParticipantObjectIdentificationForResponse(
+        ProvideAndRegisterDocumentSetRequestType request, XDRAcknowledgementType response, AssertionType assertion,
+        AuditMessageType auditMsg) {
+        // PatientParticipantObjetIdentification  and SubmissionSetParticipantObjetIdentification is same for both
+        //Request and Response in case of DS
+        auditMsg = getPatientParticipantObjectIdentification(request, auditMsg);
+        auditMsg = getSubmissionSetParticipantObjectIdentification(request, auditMsg);
+        return auditMsg;
+    }
+
+    @Override
+    protected String getServiceEventIdCodeRequestor() {
+        return DocSubmissionAuditTransformsConstants.EVENT_ID_CODE_DS_SOURCE;
```

Comment:
It would be better to move up this Event Constants getters to AbstractDocSubmissionAuditTransforms and override for DSDeferredResponse as that is using reverse constants